pontoon/translate/static/js/translate.es6

```diff
@@ -0,0 +1,499 @@
+                         onSave={this.saveTranslation.bind(this)} />
+          <div id="drag" draggable="true"></div>
+        </aside>
+      </div>
+    );
+  }
+
+  get selectedEntity() {
+    return this.state.entities[this.state.selectedIndex];
+  }
+
+  selectEntity(entityKey) {
+    let index = this.state.entities.findIndex((e) => e.pk === entityKey);
+    if (index !== -1) {
+      this.setState({
+        selectedIndex: index,
+      });
+    }
+  }
+
+  saveTranslation(string) {
+    let entity = this.selectedEntity;
+
+    entity.updateTranslation(string, this.props.locale.code).then((newEntity) => {
+      console.log('save succeeded');
+      this.setState((state, props) => {
+        let index = state.entities.findIndex((e) => e.pk === entity.pk);
+        state.entities[index] = newEntity;
+        return {entities: state.entities}
+      })
+    }).fail(() => {
+      console.log('save failed');
+    });
+  }
+
+  selectNextEntity(e) {
+    e.preventDefault();
+    if (this.state.selectedIndex < this.state.entities.length - 1) {
+      this.setState({
+        selectedIndex: this.state.selectedIndex + 1,
+      });
+    }
+  }
+
+  selectPreviousEntity(e) {
+    e.preventDefault();
+    if (this.state.selectedIndex > 0) {
+      this.setState({
+        selectedIndex: this.state.selectedIndex - 1,
+      });
+    }
+  }
+
+  toggleSidebar() {
+    this.setState((state, props) => ({
+      sidebarOpen: !state.sidebarOpen
+    }));
+  }
+}
+
+
+class LoadingScreen extends PontoonComponent {
+  render() {
+    return (
+      <div id="project-load">
+        <div className="inner">
+          <div className="animation">
+            <div></div>
+            <div></div>
+            <div></div>
+            <div></div>
+            <div></div>
+          </div>
+          <div className="text">"640K ought to be enough for anybody."</div>
+        </div>
+      </div>
+    );
+  }
+}
+
+
+/**
+ * List of entities that appears in the sidebar.
+ */
+class EntityList extends PontoonComponent {
+  render() {
+    let listBody = <h3 className="no-match"><div>ఠ_ఠ</div>No results</h3>;
+
+    if (this.props.entities.length > 0) {
+      let entities = [];
+      for (let k = 0; k < this.props.entities.length; k++) {
+        let entity = this.props.entities[k];
+        let selected = k == this.props.selectedIndex;
+
+        entities.push(
+          <EntityItem key={entity.pk}
+                      entity={entity}
+                      selected={selected}
+                      onSelectEntity={this.props.onSelectEntity} />
+        );
+      }
+
+      // Only show page mesage if we're showing an iframe.
+      let notOnPageMessage = '';
+      if (this.props.project.url) {
+        notOnPageMessage = <h2 id="not-on-page">Not on the current page</h2>;
+      }
+
+      listBody = (
+        <div className="wrapper">
+          <ul className="editables">
+            {entities}
+          </ul>
+          {notOnPageMessage}
```

Comment:
That indenting style is nice and mostly fixes my gripes with using ternary operators in JSX.